src/Parsers/Access/ParserCreateUserQuery.cpp

```diff
@@ -141,6 +147,25 @@ namespace
             }
 
             auth_data = AuthenticationData{*type};
+            if (type == AuthenticationType::SHA256_PASSWORD)
+            {
+                if (!parsed_salt.empty())
+                {
+                    auth_data.setSalt(parsed_salt);
+                }
+                else if (expect_password)
+                {
+                    ///generate and add salt here
+                    ///random generator FIPS complaint
+                    uint8_t key[32];
+                    RAND_bytes(key, sizeof(key));
+                    String salt;
+                    for (size_t i=0; i< 32; ++i)
+                        salt.append(std::to_string(key[i]));
```

Comment:
this salt looks strange to me - `std::to_string` will just generate string representation of bytes values '0'-'255' - maybe it's not a big deal, but strange... at least maybe have it hexadecimal?

Reply:
the password entered in string, before applying algorthim sha256_hash to password, appending salt to password.

Reply:
Hi Yakov, 
Salt would be generated during CREATE USER command from the CH client, So the password entered by the user will be in string. For that reason maintaining the salt in string format. Do you think any advantage converting to hexa decimal ?

Reply:
there is no particular practical reason since it's just a string - it will just look nicer, and better corresponds to convention: https://en.wikipedia.org/wiki/Salt_(cryptography)
 - `7D030F4C' looks better than `12531576', besides if, for any reason, you need to covert it back to binary, it's impossible to do it unambiguously for decimals.